src/metabase/models/field.clj

```diff
@@ -301,14 +283,22 @@
      :search
      :none)))
 
-(mi/define-batched-hydration-method with-has-field-values
-  :has_field_values
+(methodical/defmethod t2.hydrate/simple-hydrate [#_model :default #_k :has_field_values]
   "Infer what the value of the `has_field_values` should be for Fields where it's not set. See documentation for
-  [[has-field-values-options]] above for a more detailed explanation of what these values mean."
-  [fields]
-  (for [field fields]
-    (when field
-      (assoc field :has_field_values (infer-has-field-values field)))))
+  [[has-field-values-options]] above for a more detailed explanation of what these values mean.
+
+  This does one important thing: if `:has_field_values` is already present and set to `:auto-list`, it is replaced by
+  `:list` -- presumably because the frontend doesn't need to know `:auto-list` even exists?
+  See [[infer-has-field-values]] for more info."
+  [_model k field]
+  (when field
+    (assoc field k (infer-has-field-values field))))
+
+(methodical/defmethod t2.hydrate/needs-hydration? [#_model :default #_k :has_field_values]
+  "Always (re-)hydrate `:has_field_values`. This is used to convert an existing value of `:auto-list` to
```

Comment:
Here's another flex 💪 -- docstrings in `defmethod`s. And they all show up in the documentation for the multimethod itself https://github.com/camsaul/methodical#debugging 💪